[FEATURE REQUEST] Wrap errors from manager to provide pretty message

**What is your use case?**
Operate containers/versions and get higher level error messages if something goes wrong

**Is your feature request related to a problem? Please describe.**
Currently manager errors provided to cli via gRPC are very rough and dirty they need to be polished a bit by wrapping to business logic level messages. Now it looks like the following way:

```shell
→ ./dist/archived-cli_darwin_amd64_v3/archived-cli \
	--insecure \
	--endpoint=docker:5555 \
	version create docker \
		--from-yum-repo=***wrapped*** \
		--publish
WARN[2024-08-17T10:26:35+03:00] insecure flag is specified which means no TLS is in use!
panic: error creating version: rpc error: code = Unknown desc = error looking up container: sql: no rows in result set

goroutine 1 [running]:
main.main()
	/***wrapped***/github.com/teran/archived/cmd/cli/main.go:195 +0x1148
```

Which gives no idea to the user what `sql: no rows in result set` could mean.

**Describe the solution you'd like**
- Reuse similar messages from publisher wherever possible
- Provide new messages where's not

**Describe alternatives you've considered**
up to discuss

**Additional context**
n/a

--- a/manager/presenter/grpc/handlers.go
+++ b/manager/presenter/grpc/handlers.go
@@ -35,7 +35,7 @@ func New(svc service.Manager) ManageServerInterface {
 func (h *handlers) CreateContainer(ctx context.Context, in *v1.CreateContainerRequest) (*v1.CreateContainerResponse, error) {
 	err := h.svc.CreateContainer(ctx, in.GetName())
 	if err != nil {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.CreateContainerResponse{}, nil
@@ -44,7 +44,7 @@ func (h *handlers) CreateContainer(ctx context.Context, in *v1.CreateContainerRe
 func (h *handlers) RenameContainer(ctx context.Context, in *v1.RenameContainerRequest) (*v1.RenameContainerResponse, error) {
 	err := h.svc.RenameContainer(ctx, in.GetOldName(), in.GetNewName())
 	if err != nil {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.RenameContainerResponse{}, nil
@@ -53,7 +53,7 @@ func (h *handlers) RenameContainer(ctx context.Context, in *v1.RenameContainerRe
 func (h *handlers) DeleteContainer(ctx context.Context, in *v1.DeleteContainerRequest) (*v1.DeleteContainerResponse, error) {
 	err := h.svc.DeleteContainer(ctx, in.GetName())
 	if err != nil {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.DeleteContainerResponse{}, nil
@@ -62,7 +62,7 @@ func (h *handlers) DeleteContainer(ctx context.Context, in *v1.DeleteContainerRe
 func (h *handlers) ListContainers(ctx context.Context, _ *v1.ListContainersRequest) (*v1.ListContainersResponse, error) {
 	containers, err := h.svc.ListContainers(ctx)
 	if err != nil {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.ListContainersResponse{
@@ -73,7 +73,7 @@ func (h *handlers) ListContainers(ctx context.Context, _ *v1.ListContainersReque
 func (h *handlers) CreateVersion(ctx context.Context, in *v1.CreateVersionRequest) (*v1.CreateVersionResponse, error) {
 	version, err := h.svc.CreateVersion(ctx, in.GetContainer())
 	if err != nil {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.CreateVersionResponse{
@@ -84,10 +84,7 @@ func (h *handlers) CreateVersion(ctx context.Context, in *v1.CreateVersionReques
 func (h *handlers) ListVersions(ctx context.Context, in *v1.ListVersionsRequest) (*v1.ListVersionsResponse, error) {
 	versions, err := h.svc.ListAllVersions(ctx, in.GetContainer())
 	if err != nil {
-		if err == service.ErrNotFound {
-			return nil, status.Error(codes.NotFound, err.Error())
-		}
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	versionNames := []string{}
@@ -103,7 +100,7 @@ func (h *handlers) ListVersions(ctx context.Context, in *v1.ListVersionsRequest)
 func (h *handlers) DeleteVersion(ctx context.Context, in *v1.DeleteVersionRequest) (*v1.DeleteVersionResponse, error) {
 	err := h.svc.DeleteVersion(ctx, in.GetContainer(), in.GetVersion())
 	if err != nil {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.DeleteVersionResponse{}, nil
@@ -112,7 +109,7 @@ func (h *handlers) DeleteVersion(ctx context.Context, in *v1.DeleteVersionReques
 func (h *handlers) PublishVersion(ctx context.Context, in *v1.PublishVersionRequest) (*v1.PublishVersionResponse, error) {
 	err := h.svc.PublishVersion(ctx, in.GetContainer(), in.GetVersion())
 	if err != nil {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.PublishVersionResponse{}, nil
@@ -121,12 +118,12 @@ func (h *handlers) PublishVersion(ctx context.Context, in *v1.PublishVersionRequ
 func (h *handlers) CreateObject(ctx context.Context, in *v1.CreateObjectRequest) (*v1.CreateObjectResponse, error) {
 	url, err := h.svc.EnsureBLOBPresenceOrGetUploadURL(ctx, in.GetChecksum(), in.GetSize())
 	if err != nil && url == "" {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	err = h.svc.AddObject(ctx, in.GetContainer(), in.GetVersion(), in.GetKey(), in.GetChecksum())
 	if err != nil {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.CreateObjectResponse{
@@ -140,7 +137,7 @@ func (h *handlers) ListObjects(ctx context.Context, in *v1.ListObjectsRequest) (
 		if err == service.ErrNotFound {
 			return nil, status.Error(codes.NotFound, err.Error())
 		}
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.ListObjectsResponse{
@@ -154,7 +151,7 @@ func (h *handlers) GetObjectURL(ctx context.Context, in *v1.GetObjectURLRequest)
 		if err == service.ErrNotFound {
 			return nil, status.Error(codes.NotFound, err.Error())
 		}
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.GetObjectURLResponse{
@@ -165,7 +162,7 @@ func (h *handlers) GetObjectURL(ctx context.Context, in *v1.GetObjectURLRequest)
 func (h *handlers) DeleteObject(ctx context.Context, in *v1.DeleteObjectRequest) (*v1.DeleteObjectResponse, error) {
 	err := h.svc.DeleteObject(ctx, in.GetContainer(), in.GetVersion(), in.GetKey())
 	if err != nil {
-		return nil, err
+		return nil, mapServiceError(err)
 	}
 
 	return &v1.DeleteObjectResponse{}, nil
@@ -173,4 +170,12 @@ func (h *handlers) DeleteObject(ctx context.Context, in *v1.DeleteObjectRequest)
 
 func (h *handlers) Register(gs *grpc.Server) {
 	v1.RegisterManageServiceServer(gs, h)
+}
+
+func mapServiceError(err error) error {
+	switch err {
+	case service.ErrNotFound:
+		return status.Error(codes.NotFound, err.Error())
+	}
+	return status.Error(codes.Internal, err.Error())
 }

--- a/manager/presenter/grpc/handlers_test.go
+++ b/manager/presenter/grpc/handlers_test.go
@@ -22,6 +22,16 @@ func (s *manageHandlersTestSuite) TestCreateContainer() {
 	s.Require().NoError(err)
 }
 
+func (s *manageHandlersTestSuite) TestCreateContainerNotFound() {
+	s.svcMock.On("CreateContainer", "test-container").Return(service.ErrNotFound).Once()
+
+	_, err := s.client.CreateContainer(s.ctx, &v1pb.CreateContainerRequest{
+		Name: "test-container",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
+}
+
 func (s *manageHandlersTestSuite) TestRenameContainer() {
 	s.svcMock.On("RenameContainer", "old-name", "new-name").Return(nil).Once()
 
@@ -30,6 +40,17 @@ func (s *manageHandlersTestSuite) TestRenameContainer() {
 		NewName: "new-name",
 	})
 	s.Require().NoError(err)
+}
+
+func (s *manageHandlersTestSuite) TestRenameContainerNotFound() {
+	s.svcMock.On("RenameContainer", "old-name", "new-name").Return(service.ErrNotFound).Once()
+
+	_, err := s.client.RenameContainer(s.ctx, &v1pb.RenameContainerRequest{
+		OldName: "old-name",
+		NewName: "new-name",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
 }
 
 func (s *manageHandlersTestSuite) TestListContainers() {
@@ -42,6 +63,14 @@ func (s *manageHandlersTestSuite) TestListContainers() {
 	}, resp.GetName())
 }
 
+func (s *manageHandlersTestSuite) TestListContainersNotFound() {
+	s.svcMock.On("ListContainers").Return([]string{}, service.ErrNotFound).Once()
+
+	_, err := s.client.ListContainers(s.ctx, &v1pb.ListContainersRequest{})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
+}
+
 func (s *manageHandlersTestSuite) TestCreateVersion() {
 	s.svcMock.On("CreateVersion", "test-container").Return("20240102030405", nil).Once()
 
@@ -52,6 +81,16 @@ func (s *manageHandlersTestSuite) TestCreateVersion() {
 	s.Require().Equal("20240102030405", resp.GetVersion())
 }
 
+func (s *manageHandlersTestSuite) TestCreateVersionNotFound() {
+	s.svcMock.On("CreateVersion", "test-container").Return("", service.ErrNotFound).Once()
+
+	_, err := s.client.CreateVersion(s.ctx, &v1pb.CreateVersionRequest{
+		Container: "test-container",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
+}
+
 func (s *manageHandlersTestSuite) TestDeleteContainer() {
 	s.svcMock.On("DeleteContainer", "test-container").Return(nil).Once()
 
@@ -59,6 +98,16 @@ func (s *manageHandlersTestSuite) TestDeleteContainer() {
 		Name: "test-container",
 	})
 	s.Require().NoError(err)
+}
+
+func (s *manageHandlersTestSuite) TestDeleteContainerNotFound() {
+	s.svcMock.On("DeleteContainer", "test-container").Return(service.ErrNotFound).Once()
+
+	_, err := s.client.DeleteContainer(s.ctx, &v1pb.DeleteContainerRequest{
+		Name: "test-container",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
 }
 
 func (s *manageHandlersTestSuite) TestListVersions() {
@@ -80,6 +129,16 @@ func (s *manageHandlersTestSuite) TestListVersions() {
 	}, resp.GetVersions())
 }
 
+func (s *manageHandlersTestSuite) TestListVersionsNotFound() {
+	s.svcMock.On("ListAllVersions", "test-container").Return([]models.Version{}, service.ErrNotFound).Once()
+
+	_, err := s.client.ListVersions(s.ctx, &v1pb.ListVersionsRequest{
+		Container: "test-container",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
+}
+
 func (s *manageHandlersTestSuite) TestDeleteVersion() {
 	s.svcMock.On("DeleteVersion", "test-container", "test-version").Return(nil).Once()
 
@@ -90,6 +149,17 @@ func (s *manageHandlersTestSuite) TestDeleteVersion() {
 	s.Require().NoError(err)
 }
 
+func (s *manageHandlersTestSuite) TestDeleteVersionNotFound() {
+	s.svcMock.On("DeleteVersion", "test-container", "test-version").Return(service.ErrNotFound).Once()
+
+	_, err := s.client.DeleteVersion(s.ctx, &v1pb.DeleteVersionRequest{
+		Container: "test-container",
+		Version:   "test-version",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
+}
+
 func (s *manageHandlersTestSuite) TestPublishVersion() {
 	s.svcMock.On("PublishVersion", "test-container", "20240102030405").Return(nil).Once()
 
@@ -98,6 +168,17 @@ func (s *manageHandlersTestSuite) TestPublishVersion() {
 		Version:   "20240102030405",
 	})
 	s.Require().NoError(err)
+}
+
+func (s *manageHandlersTestSuite) TestPublishVersionNotFound() {
+	s.svcMock.On("PublishVersion", "test-container", "20240102030405").Return(service.ErrNotFound).Once()
+
+	_, err := s.client.PublishVersion(s.ctx, &v1pb.PublishVersionRequest{
+		Container: "test-container",
+		Version:   "20240102030405",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
 }
 
 func (s *manageHandlersTestSuite) TestCreateObject() {
@@ -115,6 +196,20 @@ func (s *manageHandlersTestSuite) TestCreateObject() {
 	s.Require().Equal("https://example.com/url", resp.GetUploadUrl())
 }
 
+func (s *manageHandlersTestSuite) TestCreateObjectNotFound() {
+	s.svcMock.On("EnsureBLOBPresenceOrGetUploadURL", "checksum", int64(1234)).Return("", service.ErrNotFound).Once()
+
+	_, err := s.client.CreateObject(s.ctx, &v1pb.CreateObjectRequest{
+		Container: "test-container",
+		Version:   "version",
+		Key:       "key",
+		Checksum:  "checksum",
+		Size:      1234,
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
+}
+
 func (s *manageHandlersTestSuite) TestListObjects() {
 	s.svcMock.On("ListObjects", "container", "version").Return([]string{"obj1", "obj2", "obj3"}, nil).Once()
 
@@ -126,6 +221,17 @@ func (s *manageHandlersTestSuite) TestListObjects() {
 	s.Require().Equal([]string{
 		"obj1", "obj2", "obj3",
 	}, resp.GetObjects())
+}
+
+func (s *manageHandlersTestSuite) TestListObjectsNotFound() {
+	s.svcMock.On("ListObjects", "container", "version").Return([]string{}, service.ErrNotFound).Once()
+
+	_, err := s.client.ListObjects(s.ctx, &v1pb.ListObjectsRequest{
+		Container: "container",
+		Version:   "version",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
 }
 
 func (s *manageHandlersTestSuite) TestGetObjectURL() {
@@ -140,6 +246,18 @@ func (s *manageHandlersTestSuite) TestGetObjectURL() {
 	s.Require().Equal("test-url", resp.GetUrl())
 }
 
+func (s *manageHandlersTestSuite) TestGetObjectURLNotFound() {
+	s.svcMock.On("GetObjectURL", "test-container", "test-version", "test-key").Return("test-url", service.ErrNotFound).Once()
+
+	_, err := s.client.GetObjectURL(s.ctx, &v1pb.GetObjectURLRequest{
+		Container: "test-container",
+		Version:   "test-version",
+		Key:       "test-key",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
+}
+
 func (s *manageHandlersTestSuite) TestDeleteObject() {
 	s.svcMock.On("DeleteObject", "test-container", "test-version", "test-key").Return(nil).Once()
 
@@ -149,6 +267,18 @@ func (s *manageHandlersTestSuite) TestDeleteObject() {
 		Key:       "test-key",
 	})
 	s.Require().NoError(err)
+}
+
+func (s *manageHandlersTestSuite) TestDeleteObjectNotFound() {
+	s.svcMock.On("DeleteObject", "test-container", "test-version", "test-key").Return(service.ErrNotFound).Once()
+
+	_, err := s.client.DeleteObject(s.ctx, &v1pb.DeleteObjectRequest{
+		Container: "test-container",
+		Version:   "test-version",
+		Key:       "test-key",
+	})
+	s.Require().Error(err)
+	s.Require().Equal("rpc error: code = NotFound desc = entity not found", err.Error())
 }
 
 // Definitions ...

--- a/repositories/metadata/metadata.go
+++ b/repositories/metadata/metadata.go
@@ -9,7 +9,10 @@ import (
 	"github.com/teran/archived/models"
 )
 
-var ErrNotFound = errors.New("not found")
+var (
+	ErrNotFound = errors.New("not found")
+	ErrConflict = errors.New("entity with given identifier already exists")
+)
 
 type Repository interface {
 	CreateContainer(ctx context.Context, name string) error

--- a/repositories/metadata/postgresql/blobs.go
+++ b/repositories/metadata/postgresql/blobs.go
@@ -4,7 +4,6 @@ import (
 	"context"
 
 	sq "github.com/Masterminds/squirrel"
-	"github.com/pkg/errors"
 )
 
 func (r *repository) CreateBLOB(ctx context.Context, checksum string, size uint64, mimeType string) error {
@@ -60,7 +59,7 @@ func (r *repository) EnsureBlobKey(ctx context.Context, key string, size uint64)
 			"size":     size,
 		}))
 	if err != nil {
-		return errors.Wrap(err, "error selecting BLOB")
+		return mapSQLErrors(err)
 	}
 
 	var blobID uint

--- a/repositories/metadata/postgresql/blobs_test.go
+++ b/repositories/metadata/postgresql/blobs_test.go
@@ -13,6 +13,10 @@ func (s *postgreSQLRepositoryTestSuite) TestBlobs() {
 	err := s.repo.CreateContainer(s.ctx, containerName)
 	s.Require().NoError(err)
 
+	_, err = s.repo.CreateVersion(s.ctx, "not-existent")
+	s.Require().Error(err)
+	s.Require().Equal(metadata.ErrNotFound, err)
+
 	versionID, err := s.repo.CreateVersion(s.ctx, containerName)
 	s.Require().NoError(err)
 

--- a/repositories/metadata/postgresql/containers.go
+++ b/repositories/metadata/postgresql/containers.go
@@ -4,7 +4,6 @@ import (
 	"context"
 
 	sq "github.com/Masterminds/squirrel"
-	"github.com/pkg/errors"
 )
 
 func (r *repository) CreateContainer(ctx context.Context, name string) error {
@@ -24,7 +23,7 @@ func (r *repository) CreateContainer(ctx context.Context, name string) error {
 func (r *repository) RenameContainer(ctx context.Context, oldName, newName string) error {
 	tx, err := r.db.BeginTx(ctx, nil)
 	if err != nil {
-		return errors.Wrap(err, "error beginning transaction")
+		return mapSQLErrors(err)
 	}
 	defer tx.Rollback()
 
@@ -38,7 +37,7 @@ func (r *repository) RenameContainer(ctx context.Context, oldName, newName strin
 
 	var containerID uint
 	if err := row.Scan(&containerID); err != nil {
-		return errors.Wrap(err, "error looking up container")
+		return mapSQLErrors(err)
 	}
 
 	_, err = updateQuery(ctx, tx, psql.
@@ -53,7 +52,7 @@ func (r *repository) RenameContainer(ctx context.Context, oldName, newName strin
 	}
 
 	if err := tx.Commit(); err != nil {
-		return errors.Wrap(err, "error committing transaction")
+		return mapSQLErrors(err)
 	}
 	return nil
 }

--- a/repositories/metadata/postgresql/containers_test.go
+++ b/repositories/metadata/postgresql/containers_test.go
@@ -1,5 +1,7 @@
 package postgresql
 
+import "github.com/teran/archived/repositories/metadata"
+
 func (s *postgreSQLRepositoryTestSuite) TestContainerOperations() {
 	s.tp.On("Now").Return("2024-01-02T01:02:03Z").Times(3)
 
@@ -15,10 +17,7 @@ func (s *postgreSQLRepositoryTestSuite) TestContainerOperations() {
 
 	err = s.repo.CreateContainer(s.ctx, "test-container9")
 	s.Require().Error(err)
-	s.Require().Equal(
-		`pq: duplicate key value violates unique constraint "containers_name_key"`,
-		err.Error(),
-	)
+	s.Require().Equal(metadata.ErrConflict, err)
 
 	list, err = s.repo.ListContainers(s.ctx)
 	s.Require().NoError(err)
@@ -30,6 +29,9 @@ func (s *postgreSQLRepositoryTestSuite) TestContainerOperations() {
 	err = s.repo.DeleteContainer(s.ctx, "test-container9")
 	s.Require().NoError(err)
 
+	err = s.repo.DeleteContainer(s.ctx, "not-existent")
+	s.Require().NoError(err)
+
 	list, err = s.repo.ListContainers(s.ctx)
 	s.Require().NoError(err)
 	s.Require().Equal([]string{
@@ -38,6 +40,10 @@ func (s *postgreSQLRepositoryTestSuite) TestContainerOperations() {
 
 	err = s.repo.RenameContainer(s.ctx, "test-container5", "and-then-there-was-the-one")
 	s.Require().NoError(err)
+
+	err = s.repo.RenameContainer(s.ctx, "not-existent", "some-name")
+	s.Require().Error(err)
+	s.Require().Equal(metadata.ErrNotFound, err)
 
 	list, err = s.repo.ListContainers(s.ctx)
 	s.Require().NoError(err)

--- a/repositories/metadata/postgresql/objects.go
+++ b/repositories/metadata/postgresql/objects.go
@@ -4,13 +4,12 @@ import (
 	"context"
 
 	sq "github.com/Masterminds/squirrel"
-	"github.com/pkg/errors"
 )
 
 func (r *repository) CreateObject(ctx context.Context, container, version, key, casKey string) error {
 	tx, err := r.db.BeginTx(ctx, nil)
 	if err != nil {
-		return errors.Wrap(err, "error beginning transaction")
+		return mapSQLErrors(err)
 	}
 	defer tx.Rollback()
 
@@ -85,7 +84,7 @@ func (r *repository) CreateObject(ctx context.Context, container, version, key, 
 	}
 
 	if err := tx.Commit(); err != nil {
-		return errors.Wrap(err, "error committing transaction")
+		return mapSQLErrors(err)
 	}
 	return nil
 }
@@ -154,7 +153,7 @@ func (r *repository) ListObjects(ctx context.Context, container, version string,
 	for rows.Next() {
 		var r string
 		if err := rows.Scan(&r); err != nil {
-			return 0, nil, errors.Wrap(err, "error decoding database result")
+			return 0, nil, mapSQLErrors(err)
 		}
 
 		result = append(result, r)
@@ -166,7 +165,7 @@ func (r *repository) ListObjects(ctx context.Context, container, version string,
 func (r *repository) DeleteObject(ctx context.Context, container, version string, key ...string) error {
 	tx, err := r.db.BeginTx(ctx, nil)
 	if err != nil {
-		return errors.Wrap(err, "error beginning transaction")
+		return mapSQLErrors(err)
 	}
 	defer tx.Rollback()
 
@@ -184,7 +183,7 @@ func (r *repository) DeleteObject(ctx context.Context, container, version string
 
 	var versionID uint
 	if err := row.Scan(&versionID); err != nil {
-		return errors.Wrap(err, "error looking up version")
+		return mapSQLErrors(err)
 	}
 
 	row, err = selectQueryRow(ctx, tx, psql.
@@ -198,7 +197,7 @@ func (r *repository) DeleteObject(ctx context.Context, container, version string
 	}
 	var okID uint
 	if err := row.Scan(&okID); err != nil {
-		return errors.Wrap(err, "error looking up version")
+		return mapSQLErrors(err)
 	}
 
 	_, err = deleteQuery(ctx, tx, psql.
@@ -212,7 +211,7 @@ func (r *repository) DeleteObject(ctx context.Context, container, version string
 	}
 
 	if err := tx.Commit(); err != nil {
-		return errors.Wrap(err, "error committing transaction")
+		return mapSQLErrors(err)
 	}
 	return nil
 }
@@ -220,7 +219,7 @@ func (r *repository) DeleteObject(ctx context.Context, container, version string
 func (r *repository) RemapObject(ctx context.Context, container, version, key, newCASKey string) error {
 	tx, err := r.db.BeginTx(ctx, nil)
 	if err != nil {
-		return errors.Wrap(err, "error beginning transaction")
+		return mapSQLErrors(err)
 	}
 	defer tx.Rollback()
 
@@ -238,7 +237,7 @@ func (r *repository) RemapObject(ctx context.Context, container, version, key, n
 
 	var versionID uint
 	if err := row.Scan(&versionID); err != nil {
-		return errors.Wrap(err, "error looking up version")
+		return mapSQLErrors(err)
 	}
 
 	row, err = selectQueryRow(ctx, tx, psql.
@@ -251,7 +250,7 @@ func (r *repository) RemapObject(ctx context.Context, container, version, key, n
 
 	var blobID uint
 	if err := row.Scan(&blobID); err != nil {
-		return errors.Wrap(err, "error looking up blob")
+		return mapSQLErrors(err)
 	}
 
 	row, err = selectQueryRow(ctx, tx, psql.
@@ -266,7 +265,7 @@ func (r *repository) RemapObject(ctx context.Context, container, version, key, n
 
 	var okID uint
 	if err := row.Scan(&okID); err != nil {
-		return errors.Wrap(err, "error looking up blob")
+		return mapSQLErrors(err)
 	}
 
 	_, err = updateQuery(ctx, tx, psql.
@@ -281,7 +280,7 @@ func (r *repository) RemapObject(ctx context.Context, container, version, key, n
 	}
 
 	if err := tx.Commit(); err != nil {
-		return errors.Wrap(err, "error committing transaction")
+		return mapSQLErrors(err)
 	}
 	return nil
 }

--- a/repositories/metadata/postgresql/objects_test.go
+++ b/repositories/metadata/postgresql/objects_test.go
@@ -71,3 +71,30 @@ func (s *postgreSQLRepositoryTestSuite) TestListObjectsErrors() {
 	s.Require().Error(err)
 	s.Require().Equal(metadata.ErrNotFound, err)
 }
+
+func (s *postgreSQLRepositoryTestSuite) TestRemapObjectErrors() {
+	// Remap with not existent container
+	err := s.repo.RemapObject(s.ctx, "not-existent", "version", "data/some-key.txt", "deadbeef")
+	s.Require().Error(err)
+	s.Require().Equal(metadata.ErrNotFound, err)
+
+	// Remap with not existent version
+	s.tp.On("Now").Return("2024-01-02T01:02:03Z").Once()
+
+	err = s.repo.CreateContainer(s.ctx, "container1")
+	s.Require().NoError(err)
+
+	err = s.repo.RemapObject(s.ctx, "not-existent", "version", "data/some-key.txt", "deadbeef")
+	s.Require().Error(err)
+	s.Require().Equal(metadata.ErrNotFound, err)
+
+	// Remap with not existent key
+	s.tp.On("Now").Return("2024-01-02T01:02:03Z").Twice()
+
+	versionID, err := s.repo.CreateVersion(s.ctx, "container1")
+	s.Require().NoError(err)
+
+	err = s.repo.RemapObject(s.ctx, "container1", versionID, "data/some-key.txt", "deadbeef")
+	s.Require().Error(err)
+	s.Require().Equal(metadata.ErrNotFound, err)
+}

--- a/repositories/metadata/postgresql/postgresql.go
+++ b/repositories/metadata/postgresql/postgresql.go
@@ -6,6 +6,7 @@ import (
 	"time"
 
 	sq "github.com/Masterminds/squirrel"
+	"github.com/lib/pq"
 	log "github.com/sirupsen/logrus"
 
 	"github.com/teran/archived/repositories/metadata"
@@ -33,9 +34,16 @@ func mapSQLErrors(err error) error {
 	switch err {
 	case sql.ErrNoRows:
 		return metadata.ErrNotFound
-	default:
-		return err
 	}
+
+	if err, ok := err.(*pq.Error); ok {
+		switch err.Code {
+		case "23505":
+			return metadata.ErrConflict
+		}
+	}
+
+	return err
 }
 
 type queryRunner interface {

--- a/repositories/metadata/postgresql/versions.go
+++ b/repositories/metadata/postgresql/versions.go
@@ -5,10 +5,10 @@ import (
 	"time"
 
 	sq "github.com/Masterminds/squirrel"
-	"github.com/pkg/errors"
 	ptr "github.com/teran/go-ptr"
 
 	"github.com/teran/archived/models"
+	"github.com/teran/archived/repositories/metadata"
 )
 
 const defaultLimit uint64 = 1000
@@ -16,7 +16,7 @@ const defaultLimit uint64 = 1000
 func (r *repository) CreateVersion(ctx context.Context, container string) (string, error) {
 	tx, err := r.db.BeginTx(ctx, nil)
 	if err != nil {
-		return "", errors.Wrap(err, "error beginning transaction")
+		return "", mapSQLErrors(err)
 	}
 	defer tx.Rollback()
 
@@ -30,7 +30,7 @@ func (r *repository) CreateVersion(ctx context.Context, container string) (strin
 
 	var containerID uint
 	if err := row.Scan(&containerID); err != nil {
-		return "", errors.Wrap(err, "error looking up container")
+		return "", metadata.ErrNotFound
 	}
 
 	versionID := r.tp().UTC().Format("20060102150405")
@@ -54,7 +54,7 @@ func (r *repository) CreateVersion(ctx context.Context, container string) (strin
 	}
 
 	if err := tx.Commit(); err != nil {
-		return "", errors.Wrap(err, "error committing transaction")
+		return "", mapSQLErrors(err)
 	}
 
 	return versionID, nil
@@ -162,7 +162,7 @@ func (r *repository) listVersionsByContainer(ctx context.Context, container stri
 		)
 
 		if err := rows.Scan(&r.Name, &r.IsPublished, &createdAt); err != nil {
-			return 0, nil, errors.Wrap(err, "error decoding database result")
+			return 0, nil, mapSQLErrors(err)
 		}
 		r.CreatedAt = time.Date(
 			createdAt.Year(), createdAt.Month(), createdAt.Day(),
@@ -179,7 +179,7 @@ func (r *repository) listVersionsByContainer(ctx context.Context, container stri
 func (r *repository) MarkVersionPublished(ctx context.Context, container, version string) error {
 	tx, err := r.db.BeginTx(ctx, nil)
 	if err != nil {
-		return errors.Wrap(err, "error beginning transaction")
+		return mapSQLErrors(err)
 	}
 	defer tx.Rollback()
 
@@ -194,7 +194,7 @@ func (r *repository) MarkVersionPublished(ctx context.Context, container, versio
 	}
 
 	if err := row.Scan(&containerID); err != nil {
-		return errors.Wrap(err, "error looking up container")
+		return metadata.ErrNotFound
 	}
 
 	_, err = updateQuery(ctx, tx, psql.
@@ -209,7 +209,7 @@ func (r *repository) MarkVersionPublished(ctx context.Context, container, versio
 	}
 
 	if err := tx.Commit(); err != nil {
-		return errors.Wrap(err, "error committing transaction")
+		return mapSQLErrors(err)
 	}
 
 	return nil
@@ -218,7 +218,7 @@ func (r *repository) MarkVersionPublished(ctx context.Context, container, versio
 func (r *repository) DeleteVersion(ctx context.Context, container, version string) error {
 	tx, err := r.db.BeginTx(ctx, nil)
 	if err != nil {
-		return errors.Wrap(err, "error beginning transaction")
+		return mapSQLErrors(err)
 	}
 	defer tx.Rollback()
 
@@ -236,7 +236,7 @@ func (r *repository) DeleteVersion(ctx context.Context, container, version strin
 
 	var versionID uint64
 	if err := row.Scan(&versionID); err != nil {
-		return errors.Wrap(err, "error looking up version")
+		return mapSQLErrors(err)
 	}
 
 	_, err = deleteQuery(ctx, tx, psql.
@@ -259,7 +259,7 @@ func (r *repository) DeleteVersion(ctx context.Context, container, version strin
 	}
 
 	if err := tx.Commit(); err != nil {
-		return errors.Wrap(err, "error committing transaction")
+		return mapSQLErrors(err)
 	}
 	return nil
 }

--- a/repositories/metadata/postgresql/versions_test.go
+++ b/repositories/metadata/postgresql/versions_test.go
@@ -90,6 +90,23 @@ func (s *postgreSQLRepositoryTestSuite) TestPublishVersion() {
 	}, list)
 }
 
+func (s *postgreSQLRepositoryTestSuite) TestPublishVersionErrors() {
+	// Not existent container
+	err := s.repo.MarkVersionPublished(s.ctx, "not-existent", "version")
+	s.Require().Error(err)
+	s.Require().Equal(metadata.ErrNotFound, err)
+
+	// Not existent version
+	s.tp.On("Now").Return("2024-07-07T10:11:12Z").Once()
+
+	err = s.repo.CreateContainer(s.ctx, "container1")
+	s.Require().NoError(err)
+
+	err = s.repo.MarkVersionPublished(s.ctx, "not-existent", "version")
+	s.Require().Error(err)
+	s.Require().Equal(metadata.ErrNotFound, err)
+}
+
 func (s *postgreSQLRepositoryTestSuite) TestListAllVersionsByContainerErrors() {
 	_, err := s.repo.ListAllVersionsByContainer(s.ctx, "test-container")
 	s.Require().Error(err)

--- a/service/service.go
+++ b/service/service.go
@@ -69,7 +69,7 @@ func newSvc(mdRepo metadata.Repository, blobRepo blob.Repository, versionsPerPag
 func (s *service) CreateContainer(ctx context.Context, name string) error {
 	err := s.mdRepo.CreateContainer(ctx, name)
 	if err != nil {
-		return errors.Wrap(err, "error creating container")
+		return mapMetadataErrors(err)
 	}
 	return nil
 }
@@ -77,7 +77,7 @@ func (s *service) CreateContainer(ctx context.Context, name string) error {
 func (s *service) RenameContainer(ctx context.Context, oldName, newName string) error {
 	err := s.mdRepo.RenameContainer(ctx, oldName, newName)
 	if err != nil {
-		return errors.Wrap(err, "error renaming container")
+		return mapMetadataErrors(err)
 	}
 	return nil
 }
@@ -111,7 +111,7 @@ func (s *service) ListPublishedVersionsByPage(ctx context.Context, container str
 	limit := s.versionsPageSize
 	totalVersions, versions, err := s.mdRepo.ListPublishedVersionsByContainerAndPage(ctx, container, offset, limit)
 	if err != nil {
-		return 0, nil, err
+		return 0, nil, mapMetadataErrors(err)
 	}
 
 	totalPages := (totalVersions / s.versionsPageSize)
@@ -138,7 +138,8 @@ func (s *service) DeleteVersion(ctx context.Context, container, id string) error
 }
 
 func (s *service) AddObject(ctx context.Context, container, versionID, key, casKey string) error {
-	return s.mdRepo.CreateObject(ctx, container, versionID, strings.TrimPrefix(key, "/"), casKey)
+	err := s.mdRepo.CreateObject(ctx, container, versionID, strings.TrimPrefix(key, "/"), casKey)
+	return mapMetadataErrors(err)
 }
 
 func (s *service) ListObjects(ctx context.Context, container, versionID string) ([]string, error) {

--- a/service/service_test.go
+++ b/service/service_test.go
@@ -25,7 +25,7 @@ func (s *serviceTestSuite) TestCreateContainer() {
 
 	err = s.svc.CreateContainer(s.ctx, "container")
 	s.Require().Error(err)
-	s.Require().Equal("error creating container: test error", err.Error())
+	s.Require().Equal("test error", err.Error())
 }
 
 func (s *serviceTestSuite) TestRenameContainer() {
@@ -40,7 +40,7 @@ func (s *serviceTestSuite) TestRenameContainer() {
 
 	err = s.svc.RenameContainer(s.ctx, "old-name", "new-name")
 	s.Require().Error(err)
-	s.Require().Equal("error renaming container: test error", err.Error())
+	s.Require().Equal("test error", err.Error())
 }
 
 func (s *serviceTestSuite) TestDeleteContainer() {

